### PR TITLE
bsp: Add esp32_s3_korvo_1 (BSP-409)

### PR DESCRIPTION
**Pull request by pborcin**
# Checklist for new Board Support package or Component

- [x] Component contains License
- [x] Component contains README.md
- [x] Project [README.md](../README.md) updated
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to CI [upload job](https://github.com/espressif/esp-bsp/blob/master/.github/workflows/upload_component.yml#L17)
- [ ] New files were added to CI build job
- [x] New BSP definitions added to [bsp_ext.py](https://github.com/espressif/esp-bsp/blob/master/examples/bsp_ext.py)
- [ ] ~~BSP was [added to SquareLine](https://github.com/espressif/esp-bsp/tree/master/SquareLine/common)~~
- [x] New BSP supports IDF v4.4.x and later releases
- [ ] ~~_Optional:_ Component contains unit tests~~
- [ ] CI passing

# Change description
Add support for esp32_s3_korvo_1

### CODE REVIEW

**Comment by pborcin** [comment]
@tore-espressif @espzav PTAL

File: bsp/esp32_s3_korvo_1/esp32_s3_korvo_1.c
Comment (espzav) [inline bsp/esp32_s3_korvo_1/esp32_s3_korvo_1.c:35]: `clk_src`, `resolution_hz`, `flags.with_dma` - not supported in IDF4.4, we should solve it. And `channel` is in IDF4.4 structure. For example: https://github.com/espressif/esp-bsp/blob/51c8afb977b169dbcb107dcbd2d6120a4926fd54/bsp/esp_bsp_generic/src/esp_bsp_generic.c#L164

**Review by espzav — COMMENTED** [review]
@pborcin Thank you very much for this PR! Good job! I left some comments. 
Please, did you test it on IDF4.4? 
I haven't HW, I cannot test it. 

Missing things: 
- [ ] Update main readme (add board with image)
- [ ] Add to tests
- [ ] Add to CI for upload component

File: bsp/esp32_s3_korvo_1/README.md
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/README.md:7]: Can we remove the references to esp-skainet for now?

File: bsp/esp32_s3_korvo_1/README.md
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/README.md:1]: Please add a picture of the board

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:18]: Please add a defintion of the buttons IO

**Review by tore-espressif — COMMENTED** [review]
Only few nitpicks for now!

**Comment by tore-espressif** [comment]
@pborcin any update about this?

File: bsp/esp32_s3_korvo_1/README.md
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/README.md:7]: Please change to absolute path, so the image is correctly rendered in https://components.espressif.com/

**Review by tore-espressif — APPROVED** [review]
Only one nitpick, LGTM!

I'll try on HW later this week

**Review by espzav — APPROVED** [review]
@pborcin Good job! Thank you.
@tore-espressif As a led_indicator and generic BSP are done, we decided to change the LEDs here to led_indicator now, before merging for not making breaking change in future. Are you agree? 
Now question is, what API we need? Is it enought only to `bsp_led_indicator_create` or do we want `bsp_led_set` or `bsp_led_rgb_set`?

**Comment by pborcin** [comment]
@tore-espressif @espzav PTAL

File: bsp/esp32_s3_korvo_1/esp32_s3_korvo_1.c
Comment (tore-espressif) [inline bsp/esp32_s3_korvo_1/esp32_s3_korvo_1.c:154]: This function is supposed to return an array of led_indicators right?

I can see that outpu parameter `led_cnt` is not set in the function, the input parameter `led_array_size` is not used, and only 1 led_indicator is initialized. Is ther esomething that I'm missing?

**Review by tore-espressif — CHANGES_REQUESTED** [review]
@pborcin I left one comment. Also, please squash your commits, before merging!

File: bsp/esp32_s3_korvo_1/esp32_s3_korvo_1.c
Comment (pborcin) [reply bsp/esp32_s3_korvo_1/esp32_s3_korvo_1.c:154]: In this case LED_STRIPS_MODE is used, with that you can control all the leds. This api is to conform with gpio mode config, where you could have more elements in the array.

Comment (tore-espressif) [reply bsp/esp32_s3_korvo_1/esp32_s3_korvo_1.c:154]: Fair enough, could you please write a note about this to the docs block of this function in .h file? Rest LGTM

**Review by tore-espressif — APPROVED** [review]

**Comment by tore-espressif** [comment]
@espzav please hit the merge button when you're done with the review, thanks!

File: bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h
Comment (espzav) [inline bsp/esp32_s3_korvo_1/include/bsp/esp32_s3_korvo_1.h:113]: Please, can you separate button section and LED section with this (it is better for reading then, if each section has this divider):
```
/**************************************************************************************************
 *
 * Buttons
 *
 **************************************************************************************************/
```
and

```
/**************************************************************************************************
 *
 * LEDs
 *
 **************************************************************************************************/
```

**Review by espzav — APPROVED** [review]
@pborcin Thank you very much for changes for led_indicator! Great work! Only one comment left. LGTM

**Pull request merged**
